### PR TITLE
Add DFA documentation for bottom friction and operator splitting

### PR DESCRIPTION
**Pull request by matthiasto**
https://docs.avaframe.org/en/adddocbottomfriction/DFAnumerics.html

### CODE REVIEW

File: docs/DFAnumerics.rst [hunk 1]
```diff
@@ -570,9 +570,12 @@ is a function of the average flow depth :math:`\overline{h}_{k}`):
 
 .. math::
     F_{k,i}^{\text{res}}
-    = - \rho_0\,A_{k}\,h^{\text{eff}}_{k}\,C_{\text{res}}\,\|\overline{\mathbf{u}}_{k}\|\,\overline{u}_{k,i}
+    = - \rho_0\,A_{k}\,h^{\text{eff}}_{k}\,C_{\text{res}}\,\|\overline{\mathbf{u}}_{k}\|^2\,\frac{\overline{u}_{k,i}}{|\overline{\mathbf{u}}_{k}\|}
     :label: resistance force
 
+Both the bottom friction and added resistance force are friction forces. The expression above represent the maximal
+friction force that can be added. This maximal force is added if the particles are flowing (if not, the friction force
+equals the driving forces) as described in :cite:`MaVi2003`.
```
Comment (fso42) [inline docs/DFAnumerics.rst:578]:
```suggestion
Both the bottom friction and resistance force are friction forces. The expression above represent the maximal
friction force that can be added. This maximal force is added if the particles are flowing. If not, the friction force
equals the driving forces.  See :cite:`MaVi2003` for more information.
```

File: docs/DFAnumerics.rst [hunk 2]
```diff
@@ -607,3 +610,69 @@ Finaly, the entrainment force reads:
 
 .. math::
     F_{k,i}^{\text{ent}} = -w_f\,(e_s+\,q_{k}^{\text{ent}}\,e_d)
+
+Adding forces
+--------------
+The different components are added following an operator splitting method.
+This means, the velocity of the particles are updated successively with the different forces.
```
Comment (fso42) [inline docs/DFAnumerics.rst:617]:
```suggestion
This means particle velocities are updated successively with the different forces.
```

File: docs/DFAnumerics.rst [hunk 3]
```diff
@@ -607,3 +610,69 @@ Finaly, the entrainment force reads:
...
+
+
+Adding artificial viscosity
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+If the viscosity option (``viscOption``) is set to 1, the artificial viscosity term is added first as described
+in :ref:`DFAnumerics:Artificial viscosity`. With ``viscOption`` set to 0, no viscosity is added. Finally, if
+``viscOption`` is set to 2, the artificial viscosity is added together with computing the SPH force (add link to description)
```
Comment (fso42) [inline docs/DFAnumerics.rst:624]:
```suggestion
If the viscosity option (``viscOption``) is set to 1, artificial viscosity is added first as described
in :ref:`DFAnumerics:Artificial viscosity`. With ``viscOption`` set to 0, no viscosity is added. Finally, if
``viscOption`` is set to 2, artificial viscosity is added during SPH force computation.  (TODO add link to description)
```
Please add the link

File: docs/DFAnumerics.rst [hunk 4]
```diff
@@ -607,3 +610,69 @@ Finaly, the entrainment force reads:
...
+
+
+Adding entrainment
+~~~~~~~~~~~~~~~~~~~
+Then, the entrainment is taken into account by adding first the component representing the loss of momentum due to
+the need to accelerate the entrained mass (:math:`- \overline{u}_{k,i}\,A^{\text{ent}}_{k}\,q^{\text{ent}}_{k}`) followed by adding the force due to the need to break and compact the
+entrained mass (:math:`F_{k,i}^{\text{ent}}`) as described in :ref:`DFAnumerics:Entrainment force`.
```
Comment (fso42) [inline docs/DFAnumerics.rst:631]:
```suggestion
Entrainment is taken into account by first adding the component representing the loss of momentum due to
acceleration of  the entrained mass :math:`- \overline{u}_{k,i}\,A^{\text{ent}}_{k}\,q^{\text{ent}}_{k}`.  Second by adding the force due to the need to break and compact the
entrained mass (:math:`F_{k,i}^{\text{ent}}`) as described in :ref:`DFAnumerics:Entrainment force`.
```
Not sure what you mean by 'need to break'

File: docs/DFAnumerics.rst [hunk 5]
```diff
@@ -607,3 +610,69 @@ Finaly, the entrainment force reads:
...
+
+
+Adding driving forces
+~~~~~~~~~~~~~~~~~~~~~~~~
+The driving forces (gravity force and lateral forces) are then taken into account. The velocity is updated explicitly.
```
Comment (fso42) [inline docs/DFAnumerics.rst:636]:
```suggestion
The driving forces -gravity force and lateral forces- are taken into account next. The velocity is updated explicitly.
```

File: docs/DFAnumerics.rst [hunk 6]
```diff
@@ -607,3 +610,69 @@ Finaly, the entrainment force reads:
...
+
+
+Adding friction forces
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Both the bottom friction and resistance forces act against the flow. There are two methods available to add these
+forces in com1DFA.
+An implicit method:
```
Comment (fso42) [inline docs/DFAnumerics.rst:643]:
```suggestion
Both the bottom friction and resistance forces act against the flow. Two methods available to add these
forces in com1DFA.
The implicit method:
```

File: docs/DFAnumerics.rst [hunk 7]
```diff
@@ -607,3 +610,69 @@ Finaly, the entrainment force reads:
...
+
+.. math::
+  \mathbf{u}_k^{new} = \frac{\mathbf{u}_k^{old}}{1 + \frac{C_{k}^{\text{fric}}\Delta t}{m_k}}
+
+where :math:`F_{k,i}^{\text{fric}} = C_{k}^{\text{fric}} u_{k,i}^{new} = F_{k,i}^{\text{res}} + F_{k,i}^{\text{bot}}`.
```
Comment (fso42) [inline docs/DFAnumerics.rst:648]: Maybe explain / spell out the parts of this equation

File: docs/DFAnumerics.rst [hunk 8]
```diff
@@ -607,3 +610,69 @@ Finaly, the entrainment force reads:
...
+
+This implicit method has a few draw-backs. First the flow does not start properly if the
+friction angle :math:`\delta` is too close to the slope angle. Second, the flow never properly stops, even if the
+particles physically should (particles keep oscillating back and force around their end position).
```
Comment (fso42) [inline docs/DFAnumerics.rst:652]:
```suggestion
particles physically should, i.e. particles keep oscillating back and force around their end position.
```

File: docs/DFAnumerics.rst [hunk 9]
```diff
@@ -607,3 +610,69 @@ Finaly, the entrainment force reads:
...
+
+The method based on :cite:`MaVi2003` should address these two issues.
```
Comment (fso42) [inline docs/DFAnumerics.rst:654]:
```suggestion
The method based on :cite:`MaVi2003` addresses these two issues.
```

File: docs/DFAnumerics.rst [hunk 10]
```diff
@@ -607,3 +610,69 @@ Finaly, the entrainment force reads:
...
+The idea is that the friction forces only modify the magnitude of velocity, not the direction (dissipation, the friction
+force can not become a driving force). Moreover, the magnitude of the friction force depends on the state of the
+particle (if it is flowing or at rest).
+The friction force is expressed:
```
Comment (fso42) [inline docs/DFAnumerics.rst:658]:
```suggestion
The idea is that the friction forces only modify the magnitude of velocity and not the direction. This means dissipation, so the friction force can not become a driving force. Moreover, the friction force magnitude depends on the particle state, i.e. if it is flowing or at rest.
The friction force is expressed:
```

**Review by fso42 — CHANGES_REQUESTED** [review]
I think this is a good way to explain different options (e.g. artif. visc)! So in my opinion we can continue this way.

Sidenote: maybe just squash the 4 commits down to 2...

File: docs/DFAnumerics.rst (hunk 3 above)
Comment (awirb) [reply docs/DFAnumerics.rst:624]: why is there a first in the first sentence - it is added as described in ref:DFA..

Comment (awirb) [reply docs/DFAnumerics.rst:624]: and maybe you could add that this is the default option?

File: docs/DFAnumerics.rst (hunk 4 above)
Comment (awirb) [reply docs/DFAnumerics.rst:631]: erode?

File: docs/DFAnumerics.rst (hunk 8 above)
Comment (awirb) [inline docs/DFAnumerics.rst:652]: back and forth

**Review by awirb — CHANGES_REQUESTED** [review]

File: docs/DFAnumerics.rst (hunk 3 above)
Comment (matthiasto) [reply docs/DFAnumerics.rst:624]: by first I mean it is the first thing we do in each time step

File: docs/DFAnumerics.rst (hunk 4 above)
Comment (matthiasto) [reply docs/DFAnumerics.rst:631]: when it is plowing I dont know if we can talk of errosion

**Pull request merged**